### PR TITLE
feat: init redesign of quickstart

### LINKED ISSUE #1092: [docs]: quickstart improve the responsiveness + design
The responsiveness for the quickstart page can be improved for various screen sizes and the elements can be improved visually by closely following the design specs. 

https://user-images.githubusercontent.com/46004116/137982995-08f8f94f-c8ec-4ba3-8cf9-39fde923ed16.mov

### PR DESCRIPTION
**Pull request by Cahllagerfeld**
## Description

Implementation of the new quickstart design
## Related Issue(s)

Fixes #1092 

Todos:

- [x] discuss how it looks when displayed on breakpoints
- [x] icons for the popular templates: On the mockup they are black, the current ones are colorful. As they are svg we can just turn them black and store them as a other 'version'. Is this the way to go?

<a href="https://gitpod.io/#https://github.com/gitpod-io/website/pull/1184"></a>

### CODE REVIEW

**Comment by mikenikles** [comment]
> discuss how it looks when displayed on breakpoints

@ChristinFrohne Could you provide a mockup for the mobile view?

> icons for the popular templates

My first thought: `class="filter grayscale"` [as per docs](https://tailwindcss.com/docs/grayscale). What do you think? We do the opposite in the "Get started now" section on the homepage.

**Comment by Cahllagerfeld** [comment]
grafik
Thats the preview from the changes of the latest commit. 
I tried it out with grayscale before, but I think the mockup contains entire black icons and not just grayscale, thats why I was asking 😄

**Comment by ChristinFrohne** [comment]
Awesome work! ✨ Thanks for picking this up @Cahllagerfeld 

Some things to watch out for:

- [ ] the logos should all appear to have the same size (this is an issue in the get started section as well)
- [ ] the color of the logos is aligned with the font color #12100C
- [ ] the logos should be transparent (e.g. Java logo) (this is an issue in the get started section as well)
- [ ] the spacing between the search bar and the templates needs to be increased
- [ ] this text should not be divided into columns
Bildschirmfoto 2021-11-03 um 09 59 00

- [ ] the text in the buttons is not centered yet
Bildschirmfoto 2021-11-03 um 10 08 24

Here is the mockup for the mobile view [[1](https://www.figma.com/file/K0VV8lc3pmlMVModYM70nj/Website?node-id=2609%3A395)]
Bildschirmfoto 2021-11-03 um 09 51 53

Let me know if anything is unclear :)

**Comment by Cahllagerfeld** [comment]
Awesome thank you! :D

**Comment by Cahllagerfeld** [comment]
Short Summary of the call with Mike from 4th November 2021:

- [ ] new icons for [Go Blue](https://commons.wikimedia.org/wiki/File:Go_Logo_Blue.svg); [Java](https://www.vectorlogo.zone/util/preview.html?image=/logos/java/java-icon.svg); [Rust](https://commons.wikimedia.org/wiki/File:Rust_programming_language_black_logo.svg)
- [ ] version with the corresponding text coloring (#12100C) are getting stored as new versions
- [ ] the problen with the centering seems to related with `inline-block` => adding custom class to it for now, rediscuss it when I finally joined
- [ ] Fix the icon issue on the main-page, in the `Launch an example workspace`-Box as well

**Comment by Cahllagerfeld** [comment]
I just came across another thing related to custom stylings
grafik
within the custom-stylings of a `btn`
```css
[class|="btn"],
.btn {
  display: inline-block;
  text-align: center;
  font-weight: 600;
  padding: 0.5rem 1.5rem;
  border-radius: 0.75em;
  box-shadow: var(--shadow-light);
  font-size: var(--btn-small);
  line-height: 16px;
  color: var(--black);
```
the line-height is set so `16px`.
In the screenshot above you can see that the text actually has a height of `18px`, which means when adding a truncate the `overflow:hidden` cuts away 2px.

How should I handle this? There are two options I think: 
1. override the line-height for these buttons
2. remove the truncate.

What do you think about this @mikenikles?

File: src/components/docs/quickstart/popular-quickstart-item.svelte [hunk 1]
```diff
@@ -4,17 +4,9 @@
...
-
-<a id="cta-button" href={path} class="btn-cta button-cta">
-  <div class="flex items-center">
-    <div class="w-12">
-      <img src={icon} alt={`${name}-icon`} width="24px" />
-    </div>
-    <span class="truncate"> More on {name}</span>
-  </div></a
->
+<a href={path} class="flex items-center text-black font-semibold btn-cta">
```
Comment (Cahllagerfeld) [inline src/components/docs/quickstart/popular-quickstart-item.svelte:7]: I found another solution for the vertical centering problem: Override all the relevant stylings from `btn-cta` with tailwind classes. I think its still a bit dirty, but will work until we redefine the component library with storybook

**Review by Cahllagerfeld — COMMENTED** [review]
One small Workaround for the vertical centering

File: package-lock.json [hunk 2]
```diff
@@ -24,7 +24,7 @@
         "github-buttons": "^2.19.1",
         "hast-util-heading-rank": "^2.1.0",
         "hastscript": "^7.0.2",
-        "husky": "^7.0.4",
+        "husky": "^7.0.0",
```
Comment (mikenikles) [inline package-lock.json:27]: @Cahllagerfeld could you merge the latest `main` into this PR branch please? This should be 7.0.4 and is fixed in `main`.

File: src/components/docs/quickstart/popular-quickstart-item.svelte (hunk 1 above)
Comment (mikenikles) [reply src/components/docs/quickstart/popular-quickstart-item.svelte:7]: Nice, I like this solution better as it aligns with our strategy to eliminate custom CSS 👍.

File: package-lock.json (hunk 2 above)
Comment (Cahllagerfeld) [reply package-lock.json]: I didnt realize my fork was behind a few commits. It sshould be fixed now 👍

**Comment by Cahllagerfeld** [comment]
awesome, thanks for merging 👍

**Pull request merged**